tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -5795,17 +5795,17 @@ SpirvInstruction *SpirvEmitter::processBinaryOp(
 
     // To evaluate this expression as an OpSpecConstantOp, we need to make sure
     // both operands are constant and at least one of them is a spec constant.
-    if (SpirvConstant *lhsValConstant = dyn_cast<SpirvConstant>(lhsVal)) {
-      if (SpirvConstant *rhsValConstant = dyn_cast<SpirvConstant>(rhsVal)) {
-        if (isAcceptedSpecConstantBinaryOp(spvOp)) {
-          if (lhsValConstant->isSpecConstant() ||
-              rhsValConstant->isSpecConstant()) {
-            auto *val = spvBuilder.createSpecConstantBinaryOp(
-                spvOp, resultType, lhsVal, rhsVal, loc);
-            val->setRValue();
-            return val;
-          }
-        }
+    if (lhsVal && rhsVal) {
```

Comment:
See the line 5812 ~ 2824. It uses `lhsVal` and `rhsVal` without null check.

Why don't we add `if (lhsVal == nullptr || rhsVal == nullptr) return nullptr;` just before this line?

Reply:
Done, thanks!